### PR TITLE
Replaced C-Style Cast with C++ Style Cast.

### PR DESCRIPTION
**Pull request by kaustubh-s1**
### Description

Please explain the changes you made, including a reference to the related issue if applicable

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by kaustubh-s1** [comment]
Hey !! I just changed the C style casts to C++ style. Please review and see if it works fine :)

File: moveit_core/robot_trajectory/include/moveit/robot_trajectory/robot_trajectory.h [hunk 1]
```diff
@@ -324,7 +324,7 @@ class RobotTrajectory
       duration_iterator++;
       return *this;
     }
-    Iterator operator++(int)
+    Iterator (operator++)static_cast<int>
```
Comment (tylerjw) [inline moveit_core/robot_trajectory/include/moveit/robot_trajectory/robot_trajectory.h:327]: I don't think this is a cast.  It is a function overload.

File: moveit_ros/perception/semantic_world/src/semantic_world.cpp [hunk 2]
```diff
@@ -374,7 +374,7 @@ SemanticWorld::generatePlacePoses(const object_recognition_msgs::msg::Table& tab
         int point_y = k * resolution * scale_factor;
         cv::Point2f point2f(point_x, point_y);
         double result = cv::pointPolygonTest(contours[0], point2f, true);
-        if ((int)result >= (int)(min_distance_from_edge * scale_factor))
+        if (static_cast<int>(result >= s)tatic_cast<int>((min_distan)ce_from_edge * scale_factor))
```
Comment (tylerjw) [inline moveit_ros/perception/semantic_world/src/semantic_world.cpp:377]:
```suggestion
        if (static_cast<int>(result) >= static_cast<int>(min_distance_from_edge) * scale_factor))
```

File: moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp [hunk 3]
```diff
@@ -530,7 +530,7 @@ void PlanningSceneMonitor::triggerSceneUpdateEvent(SceneUpdateType update_type)
 
   for (boost::function<void(SceneUpdateType)>& update_callback : update_callbacks_)
     update_callback(update_type);
-  new_scene_update_ = (SceneUpdateType)((int)new_scene_update_ | (int)update_type);
+  new_scene_update_ = (SceneUpdateType)(static_cast<int>(new)_scene_update_ | static_cast<int>(upd)ate_type);
```
Comment (tylerjw) [inline moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp:533]:
```suggestion
  new_scene_update_ = (SceneUpdateType)(static_cast<int>(new_scene_update_) | static_cast<int>(update_type));
```

File: moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp [hunk 4]
```diff
@@ -606,11 +606,11 @@ void PlanningSceneMonitor::updatePublishSettings(bool publish_geom_updates, bool
...
   PlanningSceneMonitor::SceneUpdateType event = PlanningSceneMonitor::UPDATE_NONE;
   if (publish_geom_updates)
-    event = (PlanningSceneMonitor::SceneUpdateType)((int)event | (int)PlanningSceneMonitor::UPDATE_GEOMETRY);
+    event = (PlanningSceneMonitor::SceneUpdateType)(static_cast<int>(eve)nt | static_cast<int>(Pla)nningSceneMonitor::UPDATE_GEOMETRY);
   if (publish_state_updates)
-    event = (PlanningSceneMonitor::SceneUpdateType)((int)event | (int)PlanningSceneMonitor::UPDATE_STATE);
+    event = (PlanningSceneMonitor::SceneUpdateType)(static_cast<int>(eve)nt | static_cast<int>(Pla)nningSceneMonitor::UPDATE_STATE);
   if (publish_transform_updates)
-    event = (PlanningSceneMonitor::SceneUpdateType)((int)event | (int)PlanningSceneMonitor::UPDATE_TRANSFORMS);
+    event = (PlanningSceneMonitor::SceneUpdateType)(static_cast<int>(eve)nt | static_cast<int>(Pla)nningSceneMonitor::UPDATE_TRANSFORMS);
```
Comment (tylerjw) [inline moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp:613]: Similar issues with the parenthesis in the wrong place like the other ones I pointed out above.

**Review by tylerjw — COMMENTED** [review]
I found a few minor things.  Thank you for this change!  :rocket:

File: moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp (hunk 4 above)
Comment (kaustubh-s1) [reply moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp]: I resolved this issue. Should I make a new PR for it ?  >

File: moveit_core/robot_trajectory/include/moveit/robot_trajectory/robot_trajectory.h (hunk 1 above)
Comment (kaustubh-s1) [reply moveit_core/robot_trajectory/include/moveit/robot_trajectory/robot_trajectory.h]: woops.. my bad. Undid it.

File: moveit_planners/sbpl/core/sbpl_interface/include/sbpl_interface/environment_chain3d_types.h [hunk 5]
```diff
@@ -158,7 +158,7 @@ struct EnvChain3DPlanningData
     angle_vector.resize(state_ids.size());
     for (unsigned int i = 0; i < state_ids.size(); ++i)
     {
-      if (state_ids[i] > (int)state_ID_to_coord_table_.size() - 1)
+      if (state_ids[i] > static_cast<int>(state_ID_to_coord_table_.size() - 1))
```
Comment (tylerjw) [inline moveit_planners/sbpl/core/sbpl_interface/include/sbpl_interface/environment_chain3d_types.h:161]:
```suggestion
      if (state_ids[i] > static_cast<int>(state_ID_to_coord_table_.size()) - 1)
```

File: moveit_planners/sbpl/core/sbpl_interface/src/environment_chain3d.cpp [hunk 6]
```diff
@@ -159,7 +159,7 @@ void EnvironmentChain3D::GetSuccs(int source_state_ID, std::vector<int>* succ_id
     return;
   }
 
-  if (source_state_ID > (int)planning_data_.state_ID_to_coord_table_.size() - 1)
+  if (source_state_ID > static_cast<int>(planning_data_.state_ID_to_coord_table_.size() - 1))
```
Comment (tylerjw) [inline moveit_planners/sbpl/core/sbpl_interface/src/environment_chain3d.cpp:162]:
```suggestion
  if (source_state_ID > static_cast<int>(planning_data_.state_ID_to_coord_table_.size()) - 1)
```

File: moveit_ros/perception/depth_image_octomap_updater/src/depth_image_octomap_updater.cpp [hunk 7]
```diff
@@ -263,7 +263,7 @@ void DepthImageOctomapUpdater::depthImageCallback(const sensor_msgs::msg::Image:
     {
       // wait at most 50ms
       static const double TEST_DT = 0.005;
-      const int nt = (int)(0.5 + average_callback_dt_ / TEST_DT) * std::max(1, ((int)queue_size_ / 2));
+      const int nt = static_cast<int>((0.5 + average_callback_dt_ / TEST_DT) * std::max(1, (static_cast<int>(queue_size_ / 2))));
```
Comment (tylerjw) [inline moveit_ros/perception/depth_image_octomap_updater/src/depth_image_octomap_updater.cpp:266]:
```suggestion
      const int nt = static_cast<int>((0.5 + average_callback_dt_ / TEST_DT) * std::max(1, (static_cast<int>(queue_size_) / 2)));
```

File: moveit_ros/perception/semantic_world/src/semantic_world.cpp [hunk 8]
```diff
@@ -343,9 +343,9 @@ SemanticWorld::generatePlacePoses(const object_recognition_msgs::msg::Table& tab
 
   double x_range = fabs(x_max - x_min);
   double y_range = fabs(y_max - y_min);
-  int max_range = (int)x_range + 1;
-  if (max_range < (int)y_range + 1)
-    max_range = (int)y_range + 1;
+  int max_range = static_cast<int>(x_range + 1);
+  if (max_range < static_cast<int>(y_range + 1))
+    max_range = static_cast<int>(y_range + 1);
```
Comment (tylerjw) [inline moveit_ros/perception/semantic_world/src/semantic_world.cpp:348]:
```suggestion
  int max_range = static_cast<int>(x_range) + 1;
  if (max_range < static_cast<int>(y_range) + 1)
    max_range = static_cast<int>(y_range) + 1;
```

File: moveit_ros/perception/semantic_world/src/semantic_world.cpp [hunk 9]
```diff
@@ -374,7 +374,7 @@ SemanticWorld::generatePlacePoses(const object_recognition_msgs::msg::Table& tab
         int point_y = k * resolution * scale_factor;
         cv::Point2f point2f(point_x, point_y);
         double result = cv::pointPolygonTest(contours[0], point2f, true);
-        if ((int)result >= (int)(min_distance_from_edge * scale_factor))
+        if (static_cast<int>(result) >= static_cast<int>(min_distance_from_edge) * scale_factor))
```
Comment (tylerjw) [inline moveit_ros/perception/semantic_world/src/semantic_world.cpp:377]:
```suggestion
        if (static_cast<int>(result) >= static_cast<int>(min_distance_from_edge * scale_factor))
```

File: moveit_ros/perception/semantic_world/src/semantic_world.cpp [hunk 10]
```diff
@@ -421,9 +421,9 @@ bool SemanticWorld::isInsideTableContour(const geometry_msgs::msg::Pose& pose,
 
   double x_range = fabs(x_max - x_min);
   double y_range = fabs(y_max - y_min);
-  int max_range = (int)x_range + 1;
-  if (max_range < (int)y_range + 1)
-    max_range = (int)y_range + 1;
+  int max_range = static_cast<int>(x_range + 1);
+  if (max_range < static_cast<int>(y_range + 1))
+    max_range = static_cast<int>(y_range + 1);
```
Comment (tylerjw) [inline moveit_ros/perception/semantic_world/src/semantic_world.cpp:426]:
```suggestion
  int max_range = static_cast<int>(x_range) + 1;
  if (max_range < static_cast<int>(y_range) + 1)
    max_range = static_cast<int>(y_range) + 1;
```

File: moveit_ros/visualization/rviz_plugin_render_tools/src/octomap_render.cpp [hunk 11]
```diff
@@ -169,7 +169,7 @@ void OcTreeRender::octreeDecoding(const std::shared_ptr<const octomap::OcTree>&
       bool display_voxel = false;
 
       // the left part evaluates to 1 for free voxels and 2 for occupied voxels
-      if (((int)octree->isNodeOccupied(*it) + 1) & render_mode_mask)
+      if ((static_cast<int>(octree->isNodeOccupied(*it) + 1)) & render_mode_mask)
```
Comment (tylerjw) [inline moveit_ros/visualization/rviz_plugin_render_tools/src/octomap_render.cpp:172]:
```suggestion
      if ((static_cast<int>(octree->isNodeOccupied(*it)) + 1) & render_mode_mask)
```

File: moveit_ros/visualization/rviz_plugin_render_tools/src/octomap_render.cpp [hunk 12]
```diff
@@ -203,7 +203,7 @@ void OcTreeRender::octreeDecoding(const std::shared_ptr<const octomap::OcTree>&
                 octomap::OcTreeNode* node = octree->search(key, octree_depth_);
 
                 // the left part evaluates to 1 for free voxels and 2 for occupied voxels
-                if (!(node && ((((int)octree->isNodeOccupied(node)) + 1) & render_mode_mask)))
+                if (!(node && (((static_cast<int>(octree->isNodeOccupied(node)) + 1)) & render_mode_mask)))
```
Comment (tylerjw) [inline moveit_ros/visualization/rviz_plugin_render_tools/src/octomap_render.cpp:206]:
```suggestion
                if (!(node && (((static_cast<int>(octree->isNodeOccupied(node))) + 1) & render_mode_mask)))
```

File: moveit_setup_assistant/src/tools/collision_linear_model.cpp [hunk 13]
```diff
@@ -73,7 +73,7 @@ QModelIndex CollisionLinearModel::mapToSource(const QModelIndex& proxyIndex) con
   // http://stackoverflow.com/questions/27086195/linear-index-upper-triangular-matrix
   int n = sourceModel()->columnCount();
   int k = proxyIndex.row();  // linear (row) index
-  int r = n - 2 - (int)(sqrt(-8 * k + 4 * n * (n - 1) - 7) / 2.0 - 0.5);
+  int r = n - 2 - static_cast<int>((sqrt(-8 * k + 4 * n * (n - 1) - 7) / 2.0 - 0.5));
```
Comment (tylerjw) [inline moveit_setup_assistant/src/tools/collision_linear_model.cpp:76]:
```suggestion
  int r = n - 2 - static_cast<int>(sqrt(-8 * k + 4 * n * (n - 1) - 7) / 2.0 - 0.5);
```

**Review by tylerjw — COMMENTED** [review]
I'm sorry to be so nit-picky.  Thank you for helping with this.  I'm just trying to avoid accidentally changing the existing behavior by moving what is cast to include more than just what would have been cast in with the c-style cast.

**Comment by kaustubh-s1** [comment]
> 

That's alright.

File: moveit_experimental/kinematics_cache/v2/kinematics_cache/src/kinematics_cache.cpp [hunk 14]
```diff
@@ -152,23 +152,23 @@ bool KinematicsCache::addToCache(const geometry_msgs::Pose& pose, const std::vec
...
   }
-  if (y_index >= (int)cache_size_y_ || y_index < 0)
+  if (y_index >= static_cast<int>(cache_size_y_ || y_index < 0))
   {
     ROS_DEBUG_NAMED("kinematics_cache", "Y position %f,%d lies outside grid: %d %d", pose.position.y, y_index, 0,
                     cache_size_y_);
     return false;
   }
-  if (z_index >= (int)cache_size_z_ || z_index < 0)
+  if (z_index >= static_cast<int>(cache_size_z_ || z_index < 0))
```
Comment (tylerjw) [inline moveit_experimental/kinematics_cache/v2/kinematics_cache/src/kinematics_cache.cpp:171]: Here are a few more where the parentheses change what is being cast.

File: moveit_experimental/kinematics_cache/v2/kinematics_cache/src/kinematics_cache.cpp [hunk 15]
```diff
@@ -325,14 +325,14 @@ bool KinematicsCache::readFromFile(const std::string& filename)
...
-  ROS_DEBUG_NAMED("kinematics_cache", "Read %d total points from file: %s", (int)num_solutions_vector_.size(),
+  ROS_DEBUG_NAMED("kinematics_cache", "Read %d total points from file: %s", static_cast<int>num_solutions_vector_.size(),
                   filename.c_str());
   return true;
 }
 
 bool KinematicsCache::writeToFile(const std::string& filename)
 {
-  ROS_DEBUG_NAMED("kinematics_cache", "Writing %d total points to file: %s", (int)num_solutions_vector_.size(),
+  ROS_DEBUG_NAMED("kinematics_cache", "Writing %d total points to file: %s", static_cast<int>num_solutions_vector_.size(),
```
Comment (tylerjw) [inline moveit_experimental/kinematics_cache/v2/kinematics_cache/src/kinematics_cache.cpp:335]: These are missing the parentheses.

**Comment by kaustubh-s1** [comment]
Did the changes where it wasn't really a cast 😵‍💫.

File: moveit_experimental/kinematics_cache/v2/kinematics_cache/src/kinematics_cache.cpp (hunk 14 above)
Comment (kaustubh-s1) [reply moveit_experimental/kinematics_cache/v2/kinematics_cache/src/kinematics_cache.cpp:171]: On to it. In a sec.

**Comment by tylerjw** [comment]
> Did the changes where it wasn't really a cast face_with_spiral_eyes.

Only a couple more that I found and then this needs clang-tidy-10 ran on it to fix the formatting.  We typically use a tool called `pre-commit` to do that.  To install clang-tidy-10 and pre-commit use these commands:

```
sudo apt install clang-format-10
pip3 install -U pre-commit
```

Then to run pre-commit to fix the formatting (do this in your checkout of the repo):
```
pre-commit run --all
```

Then add the changes and push it.  That will probably adjust a couple of small things.

File: moveit_experimental/kinematics_cache/v2/kinematics_cache/src/kinematics_cache.cpp (hunk 15 above)
Comment (kaustubh-s1) [reply moveit_experimental/kinematics_cache/v2/kinematics_cache/src/kinematics_cache.cpp]: Done.

**Review by tylerjw — APPROVED** [review]
This is ready to go after you run pre-commit and fix the formatting so the formatting test will pass.  Thank you for this improvement.  :smile:

**Comment by kaustubh-s1** [comment]
> 

I'm on a Mac and am facing issue installing clang-format 10. I ran pre-commit run --all and that's working fine. The only thing it says is command format 10 not found and I tried installing it using homebrew but no luck. What should I do ?

**Comment by tylerjw** [comment]
> > 
> 
> I'm on a Mac and am facing issue installing clang-format 10. I ran pre-commit run --all and that's working fine. The only thing it says is command format 10 not found and I tried installing it using homebrew but no luck. What should I do ?

For this PR, I can run it for you and update this PR.  I don't know how to install clang-format-10 on a mac.  The best idea I have is that you could maybe do that inside docker.  You'd have to create a docker image and mount your directory and then you could use the apt command it installs it and then run it.

**Comment by kaustubh-s1** [comment]
> > > 
> > 
> > 
> > I'm on a Mac and am facing issue installing clang-format 10. I ran pre-commit run --all and that's working fine. The only thing it says is command format 10 not found and I tried installing it using homebrew but no luck. What should I do ?
> 
> For this PR, I can run it for you and update this PR. I don't know how to install clang-format-10 on a mac. The best idea I have is that you could maybe do that inside docker. You'd have to create a docker image and mount your directory and then you could use the apt command it installs it and then run it.

> Thank you @tylerjw 😃.

**Pull request merged**